[1.4.0] tests fail on 32 bit (x86)

I am trying to package 1.4.0 for Debian. On the i586 platform, a few tests fail:

In `fitsio/tests/test_image_compression.py`, there are two `assert_array_equal` comparisons in `test_image_compression_read_from_osx_arm64()` and `test_image_compression_write_read_comp_to_osx_arm64()`. which fail. using `asser_array_almost_equal()` solved this.

The other failure is in `test_image_compression_big_gzip` with coef=2:
```
…
>                   out.write(out_list, names=out_names)
fitsio/tests/test_image_compression.py:670: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
fitsio/fitslib.py:858: in write
    self.write_table(
fitsio/fitslib.py:1267: in write_table
    self[-1].write(data, names=names)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = 
  file: /tmp/tmp3u64i0r5/test.fits.gz
  extension: 3
  type: BINARY_TBL
  rows: 0
  column info:
    A               ...                f8  
    C                   f8  
    D                   f8  
    E                   f8  array[10000]
data = [array([ 0.33308218, -0.51687494, -0.17298608, ...,  0.73963908,
        0.38285563,  0.59151917], shape=(10000,)), ar..., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.],
       [1., 1., 1., ..., 1., 1., 1.]], shape=(10000, 10000))]
firstrow = 0, columns = None, names = ['A', 'B', 'C', 'D', 'E'], slow = False
keys = {}
    def write(
        self, data, firstrow=0, columns=None, names=None, slow=False, **keys
    ):
…    
            if len(nonobj_arrays) > 0:
>               self._FITS.write_columns(
                    self._ext + 1,
                    nonobj_colnums,
                    nonobj_arrays,
                    firstrow=firstrow + 1,
                    write_bitcols=self.write_bitcols,
                )
E               OSError: FITSIO status = 107: tried to move past end of file
E               FITSIO status = 107: tried to move past end of file
E               FITSIO status = 107: tried to move past end of file
E               FITSIO status = 106: error writing to FITS file
E               Failed to reallocate memory (mem_write)
E               Error writing data buffer to file:
E               /tmp/tmp3u64i0r5/test.fits.gz
E               Error writing elements 3601 thru 7200 of input data array (ffpcld).
fitsio/hdu/table.py:313: OSError
```
where I don't know hos to fix this,
This is with an external cfitsio 4.6.4.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -18,6 +18,7 @@ Bug Fixes
 - Added global lock on `cfitsio` calls for non-reentrant builds.
 - Marked tests for reproducible lossy float compression as xfail for external
   cfitsio libraries.
+- Skip tests for large compressed images exceeding `2**32` bytes on 32-bit platforms.
 
 ## 1.4.0
 

--- a/fitsio/tests/test_image_compression.py
+++ b/fitsio/tests/test_image_compression.py
@@ -1,4 +1,5 @@
 import pytest
+import struct
 import sys
 import os
 import tempfile
@@ -610,6 +611,13 @@ def test_image_mem_reopen_noop():
         assert np.array_equal(rimg, img)
 
 
+@pytest.mark.skipif(
+    condition=struct.calcsize("P") * 8 == 32,
+    reason=(
+        "Cannot test writing of compressed tables "
+        "with more than 2**32 bytes on 32-bit platforms"
+    ),
+)
 @pytest.mark.slow
 @pytest.mark.parallel_threads_limit(1)
 @pytest.mark.iterations(1)

